### PR TITLE
Added support for simple dma example

### PR DESCRIPTION
**Pull request by havardh**
To test the example:
- Run with gdb
- Break at main and `print dma::DATA_DST`
- Continue to loop {} and `print dma::DATA_DST`
  The contents should now have changed

### CODE REVIEW

File: examples/dma.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#![no_std]
+#![no_main]
+#![feature(lang_items, core)]
```
Comment (sondrele) [inline examples/dma.rs]: The new version of rust requires `no_std` to be set as a feature.

File: src/dma/mod.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+pub const DMAREQ_TIMER0_UFOF: u32 = ((24 << 16) + 0);
+
+pub type FuncPtr = extern fn(channel: u32, primary: bool, user: u32);
+
+use core::intrinsics::transmute;
+use core::default::Default;
+use core::option::Option;
+use core::slice::SliceExt;
+use core::ptr;
```
Comment (sondrele) [inline src/dma/mod.rs]: Suggestion: Remove unused imports and move `use` statements to top of module.

File: src/dma/mod.rs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    pub fn configure_descriptor(&self, primary: bool, cfg: &CfgDescriptor) {
+        unsafe { DMA_CfgDescr(self.channel, primary, cfg) }
+    }
+
+    pub fn activate_auto<T>(&self, primary: bool, dst: &mut[T], src: &mut[T]) {
+        unsafe {
+
+            let (dst_n, src_n) = (dst.len(), src.len());
+
+            let n = if dst_n < src_n { dst_n } else { src_n };
```
Comment (sondrele) [inline src/dma/mod.rs]: Suggestion:
use core::cmp::min;
// ...
let n = min(dst.len(), src.len());

File: src/dma/mod.rs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    pub arb_rate: ArbiterConfig,
+    pub hprot: u8,
+}
+
+#[repr(C)]
+#[derive(Copy)]
+pub struct Descriptor {
+    pub SRCEND: u32,
+    pub DSTEND: u32,
+    pub CTRL: u32,
```
Comment (sondrele) [inline src/dma/mod.rs:84]: Generally, I think we should strive towards getting rid of all warnings. 
I think we should consider adding `#[allow(non_camel_case_types)]` for types that generate this warning, if we want them not to have snake_case.

**Comment by havardh** [comment]
Could you fix the build error on this one? Lets do it on Master and then rebase the PRs.

**Comment by sondrele** [comment]
The build-script has been updated

**Pull request merged**